### ISSUE TITLE
[Bug] Switching XRReferenceImageLibraries on ARTrackedImageManager results in strange behavior

### ISSUE THREAD

**Issue by TravestyChandler** [Unity-Technologies/arfoundation-samples#536]
**Unity bug report case number**
(Case 1261411) AR Foundation XRReferenceImageLibraries don't swap correctly

**Describe the bug**
At runtime, changing the image library of an ARTrackedImageManager results in the manager recognizing the old images, but referencing them by the names of the new ones.
Specifically, at runtime, with the ARTrackedImageManager on OR off, I am replacing the referenceLibrary with an existing one that is different from the one that is attached to it upon the Unity scene loading.

**To Reproduce**
1. Run a scene with an ARTrackedImageManager in the scene.
2. At runtime, replace the XRReferenceImageLibrary with a new one.
3. Find an image target from the new XRReferenceImageLibrary, nothing happens.
4. Find an image target from the old XRReferenceImageLibrary, the image is recognized.
5. Using Debug.Log() I can see that the image recognized is using the name of new XRReferenceImageLibrary, with the same index as the image target from the original library.

**Expected behavior**
The image from the new library should be recognized.

**Actual behavior**
Images from the original library are found.

**Smartphone (please complete the following information):**
 - Device: Google Pixel 3a
 - OS: Android 10
 - 2019.3.13f
 - AR Foundation Version 2.1.8

**Comment by tdmowrer** [comment]
Thanks for the bug report. We'll look into it.

**Comment by tdmowrer** [comment]
It looks like you've created at least one of your reference image libraries by duplicating another one and then modifying it. There is a bug related to this (see https://github.com/Unity-Technologies/arfoundation-samples/issues/596#issuecomment-696442616). You can work around this by opening your `XRReferenceImageLibrary` assets in a text editor and changing the first occurrence of `m_GuidLow` by a single digit. For example,

```
  m_Name: FreeLIbraryImageTargets
  m_EditorClassIdentifier: 
  m_GuidLow: 5428049227285229560
```

change to

```
  m_Name: FreeLIbraryImageTargets
  m_EditorClassIdentifier: 
  m_GuidLow: 1428049227285229560
```

When creating new reference image libraries, be sure to use the Assets > Create > XR > Reference Image Library option:
![Screen Shot 2020-09-21 at 5 40 43 PM](https://user-images.githubusercontent.com/4934281/93834719-ac262700-fc31-11ea-905d-0864af6d8cb7.png)

This will be fixed in a future release.

**Comment by tdmowrer** [comment]
This was [fixed in 4.1.0-preview.9](https://docs.unity3d.com/Packages/com.unity.xr.arsubsystems@4.1/changelog/CHANGELOG.html#fixes)

**Comment by Unity3D-Hardik** [comment]
Facing the same issue with AR Foundation 5.0.2 
I am trying to switch MutableRuntimeReferenceImageLibrary but after switching I can see Add Event but not the update event trigger...
